crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1088,27 +1091,24 @@ fn handle_expr_if_comment<'a>(
     CommentPlacement::Default(comment)
 }
 
-/// Moving
+/// Handles trailing comments on between the `*` of a starred expression and the
+/// expression itself. For example, grabs attaches both comments here as dangling
+/// comments on the starred expression node:
 /// ``` python
 /// call(
-///     # Leading starred comment
-///     * # Trailing star comment
+///     *  # dangling end-of-line comment
+///     # dangling own line comment
 ///     []
 /// )
 /// ```
-/// to
-/// ``` python
-/// call(
-///     # Leading starred comment
-///     # Trailing star comment
-///     * []
-/// )
-/// ```
 fn handle_trailing_expression_starred_star_end_of_line_comment<'a>(
     comment: DecoratedComment<'a>,
     starred: &'a ast::ExprStarred,
 ) -> CommentPlacement<'a> {
-    CommentPlacement::leading(starred, comment)
+    if comment.following_node().is_none() {
+        return CommentPlacement::Default(comment);
+    }
+    CommentPlacement::dangling(starred, comment)
 }
```

Comment:
So, prior to https://github.com/astral-sh/ruff/pull/6485, this method was written as:

```rust
fn handle_trailing_expression_starred_star_end_of_line_comment<'a>(
    comment: DecoratedComment<'a>,
    starred: &'a ast::ExprStarred,
) -> CommentPlacement<'a> {
    if comment.line_position().is_own_line() {
        return CommentPlacement::Default(comment);
    }

    if comment.following_node().is_none() {
        return CommentPlacement::Default(comment);
    }

    CommentPlacement::leading(starred, comment)
}
```

In that PR, it was brought up that coverage didn't seem to hit those lines, so I removed them and tests were unchanged. However, with the new tests introduced here, the current implementation gave us this formatting (for the same input mentioned in the PR summary):

```python
def test():
    (
        yield (
            # comment 1
            # comment 2
            # comment 3
            *test
        )  # comment 4
    )
```

This is _reasonable_ but a bigger deviation from Black.

The new version attaches any comments between the `*` and the starred element as dangling, and formats them around the `*` similar to Black.

Reply:
I would be fine with that, or even prefer this formatting because I believe a formatter should be opinionated about how it places comments or it doesn't solve the formatting discussions problem.

Reply:
That was my reaction too but I wasn't sure whether to bias towards this deviation or sticking to Black's formatting. I think I'll roll back.